n3fit/src/n3fit/layers/observable.py

```diff
@@ -101,13 +167,41 @@ def call(self, pdf):
             pdfs = [pdf] * len(self.padded_fk_tables)
 
         observables = []
-        for pdf, padded_fk in zip(pdfs, self.padded_fk_tables):
-            observable = self.compute_observable(pdf, padded_fk)
+        # In the case of Polarized fits, the computation of Positivity observables
+        # are much more involved. In the case of (anti-)quarks Positivity, we need
+        # to compute as final observable:
+        #      O = (q + qbar) - |Δq+Δqbar|   where Δq=Polarized quark PDF
+        # And in the case of Gluon Positivity, we need to compute:
+        #      O = g - |Δg|
+        # NOTE: Here, effectively, we are computing: (q + qbar) - |Δq| + |Δqbar|
+        # which should be fine because: |Δq+Δqbar| <= |Δq| + |Δqbar|
+        for idx, (pdf, padded_fk) in enumerate(zip(pdfs, self.padded_fk_tables)):
+            # Check if Unpolarized POS FK and convolute with the pre-computed PDF
+            if self.is_polarised_posdata() and not self.is_polarised_fktable[idx]:
+                pdf_to_convolute = self.boundary_pdf[idx]
+            else:  # Otherwsie, convolute with the usual NN PDFs
+                pdf_to_convolute = pdf
+
+            # Compute the usual convolution between the (pre-computed) PDF
+            observable = self.compute_observable(pdf_to_convolute, padded_fk)
+
+            # If it is instead a Polarized POS FK, we need to take its Absolute
+            if self.is_polarised_posdata() and self.is_polarised_fktable[idx]:
+                observable = op.multiply_minusone(op.absolute(observable))
+
             observables.append(observable)
 
         observables = self.operation(observables)
         return observables
 
+    def is_polarised_posdata(self):
+        """Check if the dataset is a Polarised Positivity dataset."""
+        # TODO: Better move the following to the new commondata input
+        if "POL" in self.fitbasis and "_POS_" in self.dataset_name:
+            # Polarised POS contains at least 2 FK tables
+            return self.nfks >= 2
```

Comment:
Yes, the positivity attribute is contained in the FK tables. However, we always need to check two conditions: (a) that the positivity dataset is polarized and (b) that the FK table is also polarized. The second condition is required because a polarized positivity dataset also contains **non-** polarized FK tables.

This is why I think that having the information that the POS dataset is polarized in the metadata would simplify things a lot. If you think that can be a solution I will extend the new commondata parser to read an optional key from the metadata?

Reply:
But that it is a positivity dataset is given by `self.dataset_name.startwith("NNPDF_POS")` and whether this is polarized should be `fktable.is_polarized()` (or whatever) but should not the fitbasis.

This will fail for instance if we happen to add non polarized composed positivities.